libs/Microsoft.MixedReality.WebRTC/Interop/PeerConnectionInterop.cs

```diff
@@ -269,31 +271,24 @@ public static void RemoteAudioFrameCallback(IntPtr userData, AudioFrame frame)
             var peer = Utils.ToWrapper<PeerConnection>(userData);
             peer.OnRemoteAudioFrameReady(frame);
         }
+        public unsafe static void SimpleStatsReportCallback(IntPtr userData, IntPtr report)
+        {
+            var tcs = Utils.ToWrapper<TaskCompletionSource<PeerConnection.StatsReport>>(userData);
+            tcs.SetResult(new PeerConnection.StatsReport(report));
+        }
 
-        public unsafe static void SimpleStatsCallback(IntPtr userData, string statsType, IntPtr statsObject)
+        public unsafe static void SimpleStatsObjectCallback(IntPtr userData, IntPtr statsObject)
         {
-            var collector = Utils.ToWrapper<PeerConnection.StatsCollector>(userData);
-            switch(statsType)
+            var list = Utils.ToWrapper<object>(userData);
+            if (list is List<PeerConnection.DataChannelStats> dataStatsList)
+            {
+                dataStatsList.Add(*(PeerConnection.DataChannelStats*)statsObject);
+            }
+            else if (list is List<PeerConnection.AudioSenderStats> audioSenderStatsList)
             {
-                case "DataChannelStats":
-                    collector.Handle(*(PeerConnection.DataChannelStats*)statsObject);
-                    break;
-                case "AudioSenderStats":
-                    collector.Handle(Marshal.PtrToStructure<PeerConnection.AudioSenderStats>(statsObject));
-                    break;
-                case "AudioReceiverStats":
-                    collector.Handle(Marshal.PtrToStructure<PeerConnection.AudioReceiverStats>(statsObject));
-                    break;
-                case "VideoSenderStats":
-                    collector.Handle(Marshal.PtrToStructure<PeerConnection.VideoSenderStats>(statsObject));
-                    break;
-                case "VideoReceiverStats":
-                    collector.Handle(Marshal.PtrToStructure<PeerConnection.VideoReceiverStats>(statsObject));
-                    break;
-                case "TransportStats":
-                    collector.Handle(*(PeerConnection.TransportStats*)statsObject);
-                    break;
+                audioSenderStatsList.Add(Marshal.PtrToStructure<PeerConnection.AudioSenderStats>(statsObject));
```

Comment:
Did you figure out the difference between direct pointer cast and `PtrToStructure`?

Reply:
Pointer cast only compiles if the struct is blittable (makes sense).